patches/tModLoader/Terraria/Audio/SoundEngine.cs.patch

```diff
@@ -9,6 +18,75 @@
  		}
  
  		public static void Load(IServiceProvider services) {
+@@ -60,13 +_,20 @@
+ 			PlaySound(type, (int)position.X, (int)position.Y, style);
+ 		}
+ 
+-		public static SoundEffectInstance PlaySound(LegacySoundStyle type, Vector2 position) => PlaySound(type, (int)position.X, (int)position.Y);
++		public static SoundEffectInstance PlaySound(ISoundStyle type, Vector2 position) => PlaySound(type, (int)position.X, (int)position.Y);
```

Comment:
Proposal to add an parameter `bool? farAway` specified between for panning in a static way (far away target located in the direction of normalized position difference on the frame this is called) and have `null` be the default used for vanilla effects, so that a mod can inject here and detect if the mod author forced on or off dynamic panning or didn't care (null is the same as true in this case to preserve vanilla behaviour, but can be easily be detoured to be false)